src/commands/site-list.php

```diff
@@ -167,32 +212,158 @@ function( $site ) {
 		}
 	}
 
-	protected function count_sites( $site_list, $term ) {
+	protected function eval_which_host( $site, $pressable_sites ) {
+		if ( true === $site->is_wpcom_atomic ) {
+			$server = 'Atomic';
+		} elseif ( true === $site->jetpack ) {
+			if ( in_array( parse_url( $site->URL, PHP_URL_HOST ), $pressable_sites ) ) {
+				$server = 'Pressable';
+			} else {
+				$server = 'Other';
+			}
+		} else {
+			$server = 'Simple'; // Need a better way to determine if site is simple. eg. 410'd Jurrasic Ninja sites will show as Simple.
+		}
+		return $server;
+	}
+
+	protected function filter_public_sites( $site_list ) {
+		$filtered_site_list = array();
+		foreach ( $site_list as $site ) {
+			if ( '' === $site[9] && '' === $site[4] && '' === $site[5] && ( 'is_subsite' !== $site[7] || '' !== $site[3] ) ) {
+				if ( '' === $site[2] || ( '' !== $site[2] && '' !== $site[3] ) ) {
+					$filtered_site_list[] = array(
+						$site[0],
+						$site[1],
+						$site[8],
+						$site[6],
+					);
+				}
+			}
+		}
+		return $filtered_site_list;
+	}
+
+	protected function eval_site_list( $site_list, $audit_type ) {
+		$audit_site_list = array();
+		foreach ( $site_list as $site ) {
+			if ( 'no-staging' === $audit_type && false !== strpos( $site[1], 'staging' ) ) {
+				continue;
+			}
+			if ( 'full' !== $audit_type && 'no-staging' !== $audit_type && ! in_array( $audit_type, $site, true ) ) {
+				continue;
+			}
+			if ( '' === $site[9] && '' === $site[4] && '' === $site[5] && ( 'is_subsite' !== $site[7] || '' !== $site[3] ) ) {
+				if ( '' === $site[2] || ( '' !== $site[2] && '' !== $site[3] ) ) {
+					$result = 'PASS';
+				} else {
+					$result = 'FAIL';
+				}
+			} else {
+				$result = 'FAIL';
+			}
+			$audit_site_list[] = array(
+				$site[0],
+				$site[1],
+				$site[2],
+				$site[3],
+				$site[4],
+				$site[5],
+				$site[7],
+				$site[9],
+				$site[6],
+				$result,
+				$site[8],
```

Comment:
This would be easier to read/understand/maintain if it uses associative keys instead of numeric ones. 

The if clauses would be much more self-explanatory. Otherwise, we need to continuously check the `$full_site_list` array as a reference to know what's exactly in each position. 

In the case that we add an extra column to the `$full_site_list` array (at position `[4]` for example), we would also need to update all the references to the rest of the items (i.e: `$site[4]` `$site[5]`, etc to the new key `$site[5]`, `$site[6]`, etc).

Reply:
This is partially complete in https://github.com/Automattic/team51-cli/pull/159/commits/e7371e0db23eb074ba3bcbc1b604d2ac1e952780

I still need to make the `$audit_site_list` an associative array to be able to implement the remaining two comments.

Reply:
Completed in https://github.com/Automattic/team51-cli/pull/159/commits/f7dcce423becc7731b6acd5c0c6b98dcaf7840c0